modules/ml/sagemaker-custom-kernel/scripts/build_docker.sh

```diff
@@ -0,0 +1,11 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+#!/usr/bin/env bash
+
+aws ecr get-login-password --region $AWS_DEFAULT_REGION | docker login --username AWS --password-stdin $AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com
+echo Building the Docker image...
+cache_image=$IMAGE_URI
+docker pull ${cache_image} 2>&1 > /dev/null || true
+cd docker/$SEEDFARMER_PARAMETER_SAGEMAKER_IMAGE_NAME && docker build --progress plain --cache-from=${cache_image} -t $IMAGE_URI .
+docker push $IMAGE_URI
```

Comment:
Instead of going with the traditional way of building docker images via CLI, you could use CDK way of building docker images. Ref: https://github.com/srinivasreddych/demo-with-mps3-reinvent23/blob/main/modules/pre-processing/image-extraction/stack.py#L68

This will look for a dockerfile, build and push it to ECR and you can avoid the build script and build commands in the deployspec.yaml

Reply:
Ah, nice, didn't know about this. Thanks!

Reply:
Refactored to use this. Thanks Srini!